### ISSUE TITLE
Very fast agent learning and training

### ISSUE THREAD

**Issue by makus82** [OpenApoc/OpenApoc#613]
Very fast agent learning and training, and i understand that it maybe for testing reasons but when realease will come its will be an issue.
Learning speed should be reduced in 4 times)
(notice it at Quickmind streams and he also notice it)

**Comment by FilmBoy84** [comment]
I'm thinking there should be a variable XP modifier for how fast agents gain experience
The default should be as per original game, and scalable up or down in percentage increments or whatever works for the code

**Comment by Quickmind01** [comment]
Yeah, as much as I loved it, a handful of "training missions" will give you super soldiers. Perhaps later, something could be added to enhance stat growth from missions? Like a passive item or something.

**Comment by FilmBoy84** [comment]
This appears to be caused, in part, by agents shooting civilians and getting rewarded for it

See #305  and #410

**Comment by FilmBoy84** [comment]
This issue is still not resolved

Agents that live more than a handful of missions will train and gain experience rapidly

![image](https://user-images.githubusercontent.com/33913088/230032185-2b6e5436-44ba-49c7-95e7-b694c341d73e.png)

I cannot find the PR that resolves this, and matches the original game mechanism for training, so reopening

**Comment by FilmBoy84** [comment]
As reported by @makus82 in the now merged #1248 PR for rendering Agent Award feedback, there is also a string for overall percentage improvement to an agent's stats from training.

This issue will likely need resolving before that functionality can be restored as per the OG

![image](https://github.com/OpenApoc/OpenApoc/assets/33913088/a96e703e-fd64-43dc-b1da-49567bc3d969)

**Comment by stewartmatheson** [comment]
From https://www.ufopaedia.org/index.php/Training_(Apocalypse)

> Every midnight a soldier in combat training has a chance of improving combat stats (Health, Accuracy, Reactions, TU, Stamina, and Strength) and a soldier in psionic training has a chance of improving psionic stats (Psi-energy, Psi-attack and Psi-defense). 

Looking at https://github.com/OpenApoc/OpenApoc/blob/638462059dfea60d3f53cb40fc1bec0864e0263e/game/state/shared/agent.cpp#L1169 it would seem that training in OpenApoc updates more often than that. I've spent some time debugging and confirmed that training buffs are being applied every 4 hours or so.  If the above guide is correct we could just move the training so that it will execute once a day instead.

**Comment by FilmBoy84** [comment]
Many thanks for narrowing down this issue Stewart

For training, the UFOPaedia appears correct - however, i cannot see the calculations for how "effectiveness" of training is applied - for example when a training facility is under or over occupancy?

Under should train agents faster and over with a penalty. We made need some reversing of the OG to ascertain how this is done

However, agents, hybrids AND Androids can get stat boosts from battlescape actions (this is indeed the only way to improve android stats in the OG, albeit at a lower rate than the other two types)

I do not believe we have a concrete calculation from the OG yet as to how this experience is applied and, IIRC, it is applied on the end of each battle's conclusion - allowing multiple stat improvements in a day if an agent is present at multiple skirmishes
